### ISSUE TITLE
Includes do not seem to work correctly

### ISSUE THREAD

**Issue by tallero** [python-poetry/poetry#9904]
### Description

Hello,
I'm opening this issue just for conscience as if I really needed to fix these includes I would do it in the downstream distribution recipe script but anyway.

I can't seem to make the include/exclude fields work for a cythonized module using the reported syntax from the documentation.

This is the [pyproject.toml](https://github.com/themartiancompany/aioetherscan/blob/master/pyproject.toml) file I've been using. 

This is the [build.log](https://github.com/user-attachments/files/18142719/build.log), this is [poetry.config.log](https://github.com/user-attachments/files/18142722/poetry.config.log).

### Workarounds

Delete the extra files from the downstream distribution build recipe.

### Poetry Installation Method

system package manager (eg: dnf, apt etc.)

### Operating System

Life / DogeOS

### Poetry Version

1.8.4

### Poetry Configuration

```bash session
It's attached above.
```

### Example pyproject.toml

```TOML
I've linked it in the description above.
```

### Poetry Runtime Logs

```bash session
I'm adding it as an attachment above because it seems too long for the github issue.
```

**Comment by dimbleby** [comment]
You are using the wrong build-backend, please see the docs

**Comment by tallero** [comment]
I've switched back to setuptools and ditched pep517 entirely.

**Comment by dimbleby** [comment]
If you are here for help then please use the right build backend - and make clear what happened vs what you expected to happen.  All you have said so far is that it is something to do with include/exclude: but we have no way of knowing what you think poetry is actually doing wrong.

If you are here only to vent then: job done, now please close the issue.

**Comment by tallero** [comment]
I gave you the sources and the log. You have all you need to reproduce the issue and know what poetry is doing wrong sir.

**Comment by dimbleby** [comment]
please say what happened vs what you expected to happen.

**Comment by tallero** [comment]
Fourth line

>I can't seem to make the include/exclude fields work

Ninth line

>(workaround is to) Delete the extra files from the downstream distribution build recipe.

Sixth line

>here is the code you can use to reproduce the issue.

**Comment by Secrus** [comment]
Please tone down @tallero, @dimbleby

**Comment by tallero** [comment]
@Secrus im sorry but certain people seem to easily mistake community sunday nights bug reports for help requests.

**Comment by tallero** [comment]
@dimbleby why have you even start reading the log without having looked at the `pyproject.toml`? Imagine if that was empty!
Of course there is something I expected to included/excluded differently, and it's literally the only notable thing in that file.
![image](https://github.com/user-attachments/assets/31c7928e-53a4-400c-bd1d-0c67623871f9)

@Secrus
So the issue is I was using the refactored >=1.x include format with a backend from the 0.1x times?
More in general I'm really puzzled as to how PEP517 has solved anything in regard to package building, as since the switch `setuptools` requires `build` to build and `build` requires `setuptools` to build, so in the end packagers still require an external Python distribution to bootstrap core packages and still version-independent modules are installed into version-specific locations like `/usr/lib/python<version>/site-packages`, so requiring a new re-build each X months, so that in the end the greatest maintainable python libraries seem to be those which install themselves into their own locations at `/usr/lib/<program>` and depend on external build systems like GNU Make, meson and the likes.

So anyway I've edited the `pyproject.toml` file to use `poetry.core.masonry.api` instead of `poetry.masonry.api` but still the `pyx` files are included in both the source distribution and the wheel.
Also if I try to run `python -m build --verbose --wheel --no-isolation` in the directory there is no output at all.

Again, here is the [pyproject.toml](https://github.com/themartiancompany/aioetherscan/blob/master/pyproject.toml), here is the master branch targeting [build recipe](https://github.com/themartiancompany/python-aioetherscan-git-ur).

The log is being produced right now, but really I struggle to see at all how poetry may be fit at all for system packaging, since despite the various flags I have switched I can't seem to make it use the locally installed (and verified) dependencies instead of (unsecurely) retrieving and rebuilding them from pypi.

**Comment by tallero** [comment]
Ok, so now it seems at least the `.c` files have been excluded from the wheel, but still the `.pyx` files are there.
Here is the new [build.log](https://github.com/user-attachments/files/18153710/build.log).

**Comment by dimbleby** [comment]
> but still the pyx files are included in both the source distribution and the wheel.

I infer that this is what you find unexpected.  I wish you would have said so sooner!

It looks to me as though https://github.com/python-poetry/poetry-core/blob/4749d63c822147b0dbaa82033c02628f92c97200/src/poetry/core/masonry/builders/builder.py#L94 expects `exclude` to be a plain list of strings, a dictionary of form `{"path": "this", "format": "that"}` simply is not supported.

It's not clear to me whether that's intended or not.  The docs "clarification" at #8852 kind implied that it was expected to work - but perhaps it never was.

**Comment by dimbleby** [comment]
... which makes this a duplicate of #3380, see in particular https://github.com/python-poetry/poetry/issues/3380#issuecomment-731273686

**Comment by tallero** [comment]
@dimbleby the [statement](https://python-poetry.org/docs/pyproject/) "exclude *defaults* to both `sdist` and `wheel`" induced me to think it was possible to select the destination, as indeed one may want to exclude the `.pyx` only from the wheel. 
![image](https://github.com/user-attachments/assets/70a4367e-57ee-4115-90d8-107a94779433)

**Comment by dimbleby** [comment]
right, that's exactly the text that was added in #8852, I agree that it is misleading

I expect that a pull request correcting it to say that exclude _always_ applies to both formats would be welcome

**Comment by tallero** [comment]
@dimbleby I would be glad to fix that issue for the community, but right now I don't think I have the time to as I have to find a way to save myself from starvation before the end of the month, so for now I think I will stick to use setuptools.
